src/ecs/src/world.rs

```diff
@@ -27,23 +28,43 @@ impl World {
     /// Creates a new entity in the world and returns a handle to it.
     pub fn create_entity(&mut self) -> Entity {
         let id = self.entities.create();
-        self.components.insert(id.clone(), AnyVec::new());
         id
     }
 
     /// Destroys a given entity and deallocates its components.
     pub fn destroy_entity(&mut self, entity: Entity) {
-        self.components.remove(&entity);
         self.entities.destroy(entity);
     }
 
     /// Attaches a component to an entity.
     pub fn insert_component<T: Any>(&mut self, entity: Entity, comp: T) {
         if self.entities.is_alive(entity) {
-            if let Some(c) = self.components.get_mut(&entity) {
-                c.push(comp);
+            let t = TypeId::of::<(Entity, T)>();
+            if let Some(c) = self.components.get_mut(&t) {
+                c.push((entity, comp));
+                return;
             }
+            let mut vec = DynVec::new::<(Entity, T)>();
+            vec.push((entity, comp));
+            self.components.insert(t, vec);
+        }
+    }
+
+    /// Returns ith component of selected type
+    pub fn component<T: Any>(&self, index: usize) -> Option<&(Entity, T)> {
+        if let Some(c) = self.components.get(&TypeId::of::<(Entity, T)>()) {
```

Comment:
You should probably use the following, for the sake of consistency with `insert_component()`:

```rust
let t = TypeId::of::<(Entity, T)>();
if let Some(c) = self.components.get(&t) {
```